src/store/batch-ingredients/batch-ingredients.types.ts

```diff
@@ -0,0 +1,48 @@
+import { BatchId } from "../batches/batches.types";
+import { IngredientId } from "../ingredients/ingredients.types";
```

Comment:
The other tables all have a very similar structure and content to the initial products files I asked many questions about. But here in batch ingredients something more complex is required. We import types we have defined elsewhere.

Reply:
Indeed, this is essentially because a batch has a foreign key, it use ingredient_id which references the ingredient table